### PR TITLE
Added some fallbacks to avoid panics

### PR DESCRIPTION
**Pull request by milapsheth**

### CODE REVIEW

File: src/gmp_backend.rs [hunk 1]
```diff
@@ -223,11 +237,20 @@ impl Bn {
...
     pub fn to_bytes(&self) -> Vec<u8> {
+        debug_assert!(self >= &Self::zero());
+
+        if self < &Self::zero() {
+            return vec![0];
+        }
+
         let mut s = self.0.to_str_radix(16);
+
         if s.len() & 1 == 1 {
```
Comment (ggutoski) [inline src/gmp_backend.rs:249]: I know this isn't your code, but this `if` statement could benefit from an explanation.  Perhaps we can mention it in an upstream PR.

File: src/gmp_backend.rs [hunk 2]
```diff
@@ -223,11 +237,20 @@ impl Bn {
...
             s = format!("0{}", s);
         }
-        hex::decode(&s).unwrap()
+
+        // TODO: Use a bad default instead of panic. This should be replaced with a nicer function
```
Comment (ggutoski) [inline src/gmp_backend.rs]: I don't understand this comment.  Aren't you already using a "bad" default (ie. 0)?  What would a "nicer function" accomplish?

**Review by ggutoski — APPROVED** [review]
Please read comments then merge at your discretion.

File: src/gmp_backend.rs (hunk 1 above)
Comment (milapsheth) [reply src/gmp_backend.rs:249]: Sure

File: src/gmp_backend.rs (hunk 2 above)
Comment (milapsheth) [reply src/gmp_backend.rs]: Sorry, by "this" I meant the `to_bytes` function should be replaced with one that can't panic.

**Review by ggutoski — APPROVED** [review]

**Pull request merged**